emain/emain-window.ts

```diff
@@ -301,9 +303,9 @@ export class WaveBrowserWindow extends BaseWindow {
 
         // If the workspace is already owned by a window, then we can just call SwitchWorkspace without first prompting the user, since it'll just focus to the other window.
         const workspaceList = await WorkspaceService.ListWorkspaces();
-        if (!workspaceList?.find((wse) => wse.workspaceid === workspaceId)?.windowid) {
-            const curWorkspace = await WorkspaceService.GetWorkspace(this.workspaceId);
-            if (isNonEmptyUnsavedWorkspace(curWorkspace)) {
+        if (!workspaceList?.find((wse) => wse.workspaceid === workspaceId)?.windowid) {		
+            const curWorkspace = await WorkspaceService.GetWorkspace(this.workspaceId);  /* potential bug here, curWorkspace undefined after deleting previous workspace */
```

Comment:
Hmm, this bug seems strange, you're still noticing this after your backend change?

Reply:
![image](https://github.com/user-attachments/assets/ad45648b-9b2c-406c-9da0-d5bb60fdaffe)
![image](https://github.com/user-attachments/assets/8db2960f-7014-47aa-82bd-c3b28afb5e98)

The bug occurs both in the backend and in electron, because whenever we delete the current workspace we still try to access the 'currently deleted workspace' from the window object.